samza-kafka/src/test/scala/org/apache/samza/config/TestKafkaConfig.scala

```diff
@@ -195,25 +195,136 @@ class TestKafkaConfig {
+    props.setProperty("task.checkpoint.replication.factor", "4")
+
+    val mapConfig = new MapConfig(props.asScala.asJava)
+    val kafkaConfig = new KafkaConfig(mapConfig)
+    assertEquals("kafka-system", kafkaConfig.getCheckpointSystem.orNull)
+    assertEquals("4", kafkaConfig.getCheckpointReplicationFactor.orNull)
+  }
+
+  @Test
+  def testCheckpointReplicationFactorWithSystemDefault() {
+    props.setProperty(JobConfig.JOB_DEFAULT_SYSTEM, "kafka-system-39-not38-not40-39")
+    props.setProperty("systems.kafka-system-39-not38-not40-39.default.stream.replication.factor", "8")
+    props.setProperty("systems.kafka-system-39-not38-not40-39.default.stream.segment.bytes", "8675309")
+
+    val mapConfig = new MapConfig(props.asScala.asJava)
+    val kafkaConfig = new KafkaConfig(mapConfig)
+    assertEquals("kafka-system-39-not38-not40-39", kafkaConfig.getCheckpointSystem.orNull)
+    assertEquals("8", kafkaConfig.getCheckpointReplicationFactor.orNull)
+    assertEquals(8675309, kafkaConfig.getCheckpointSegmentBytes)
+  }
+
+  @Test
+  def testCheckpointReplicationFactorWithSystemOverriddenDefault() {
+    // Defaults
+    props.setProperty(JobConfig.JOB_DEFAULT_SYSTEM, "kafka-system-39-not38-not40-39")
+    props.setProperty("systems.kafka-system-39-not38-not40-39.default.stream.replication.factor", "8")
+    props.setProperty("systems.kafka-system.default.stream.segment.bytes", "8675309")
+
+    // Overrides
+    props.setProperty(KafkaConfig.CHECKPOINT_SYSTEM, "kafka-system")
+    props.setProperty(KafkaConfig.CHECKPOINT_REPLICATION_FACTOR, "4")
+
+    val mapConfig = new MapConfig(props.asScala.asJava)
+    val kafkaConfig = new KafkaConfig(mapConfig)
+    assertEquals("kafka-system", kafkaConfig.getCheckpointSystem.orNull)
+    assertEquals("4", kafkaConfig.getCheckpointReplicationFactor.orNull)
+    assertEquals(8675309, kafkaConfig.getCheckpointSegmentBytes)
+  }
+
+  @Test
+  def testCoordinatorReplicationFactor() {
+    val emptyConfig = new KafkaConfig(new MapConfig())
+    assertEquals("3", emptyConfig.getCoordinatorReplicationFactor)
+    assertNull(new JobConfig(new MapConfig()).getCoordinatorSystemNameOrNull)
+
+    props.setProperty(JobConfig.JOB_COORDINATOR_SYSTEM, "kafka-system")
+    props.setProperty(KafkaConfig.JOB_COORDINATOR_REPLICATION_FACTOR, "4")
+
+    val mapConfig = new MapConfig(props.asScala.asJava)
+    val kafkaConfig = new KafkaConfig(mapConfig)
+    val jobConfig = new JobConfig(mapConfig)
+    assertEquals("kafka-system", jobConfig.getCoordinatorSystemNameOrNull)
+    assertEquals("4", kafkaConfig.getCoordinatorReplicationFactor)
+  }
+
+  @Test
+  def testCoordinatorReplicationFactorWithSystemDefault() {
+    props.setProperty(JobConfig.JOB_DEFAULT_SYSTEM, "kafka-system-39-not38-not40-39")
+    props.setProperty("systems.kafka-system-39-not38-not40-39.default.stream.replication.factor", "8")
+    props.setProperty("systems.kafka-system-39-not38-not40-39.default.stream.segment.bytes", "8675309")
```

Comment:
nit: 
what does the `kafka-system-39-not38-not40-39 ` test? Is there parsing logic that relies on a lot of hyphens? If not, a shorter `system name` might be nicer :-)